src/python/dxpy/templating/templates/nextflow/src/nextflow.sh

```diff
@@ -110,86 +242,129 @@ dx_path() {
-    export NXF_HOME=/opt/nextflow
-    export NXF_UUID=$(uuidgen)
-    export NXF_ANSI_LOG=false
-    export NXF_EXECUTOR=dnanexus
-    export NXF_PLUGINS_DEFAULT=nextaur@1.1.0
-    export NXF_DOCKER_LEGACY=true
-    #export NXF_DOCKER_CREDS_FILE=$docker_creds_file
-    #[[ $scm_file ]] && export NXF_SCM_FILE=$(dx_path $scm_file 'Nextflow CSM file')
-
-    mkdir -p /home/dnanexus/out/output_files
-    cd /home/dnanexus/out/output_files
-    
-    generate_runtime_config
-    
-    NEXTFLOW_CMD="nextflow \
-      ${TRACE_CMD} \
-      $nextflow_top_level_opts \
-      ${RUNTIME_CONFIG} \
-      -log ${LOG_NAME} \
-      run @@RESOURCES_SUBPATH@@ \
-      @@PROFILE_ARG@@ \
-      -name run-${NXF_UUID} \
-      $nextflow_run_opts \
-      $nextflow_pipeline_params \
-      @@REQUIRED_RUNTIME_PARAMS@@
+  if [[ $preserve_cache == true ]]; then
+    dx set_properties "$DX_JOB_ID" nextflow_executable="$EXECUTABLE_NAME" nextflow_session_id="$NXF_UUID" nextflow_preserve_cache="$preserve_cache"
+  fi
+
+  if [[ $preserve_cache == true && -n $resume ]]; then
+    check_running_jobs
+  fi
+
+  RESUME_CMD=""
+  if [[ -n $resume ]]; then
+    restore_cache
+  fi
+
+  setup_workdir
+  export NXF_WORK
+
+  # for optional inputs, pass to the run command by using a runtime config
+  # TODO: better handling inputs defined in nextflow_schema.json
+  RUNTIME_CONFIG_CMD=""
+  generate_runtime_config
+
+  # set beginning timestamp
+  BEGIN_TIME="$(date +"%Y-%m-%d %H:%M:%S")"
+
+  # execution starts
+  NEXTFLOW_CMD="nextflow \
+    ${TRACE_CMD} \
+    $nextflow_top_level_opts \
+    ${RUNTIME_CONFIG_CMD} \
+    -log ${LOG_NAME} \
+    run @@RESOURCES_SUBPATH@@ \
+    @@PROFILE_ARG@@ \
+    -name $DX_JOB_ID \
+    $RESUME_CMD \
+    $nextflow_run_opts \
+    $nextflow_pipeline_params \
+    @@REQUIRED_RUNTIME_PARAMS@@
       "
 
-    trap on_exit EXIT
-    echo "============================================================="
-    echo "=== NF work-dir : ${DX_WORK}"
-    echo "=== NF log file : ${DX_LOG}"
-    echo "=== NF cache    : $DX_PROJECT_CONTEXT_ID:/.nextflow/cache/$NXF_UUID"
-    echo "=== NF command  :" $NEXTFLOW_CMD
-    echo "============================================================="
-
-    $NEXTFLOW_CMD & NXF_EXEC_PID=$!
-    
-    # forwarding nextflow log file to job monitor
-    set +x
-    if [[ $debug == true ]] ; then
-      touch $LOG_NAME
-      tail --follow -n 0 $LOG_NAME -s 60 >&2 & LOG_MONITOR_PID=$!
-      disown $LOG_MONITOR_PID
-      set -x
-    fi
-    
-    wait $NXF_EXEC_PID
-    ret=$?
-    exit $ret
-}
+  trap on_exit EXIT
+  echo "============================================================="
+  echo "=== NF projectDir   : @@RESOURCES_SUBPATH@@"
+  echo "=== NF session ID   : ${NXF_UUID}"
+  echo "=== NF log file     : dx://${DX_JOB_OUTDIR%/}/${LOG_NAME}"
+  if [[ $preserve_cache == true ]]; then
+    echo "=== NF cache folder : dx://${DX_CACHEDIR}/${NXF_UUID}/"
+  fi
+  echo "=== NF command      :" $NEXTFLOW_CMD
```

Comment:
Currently the formatting for the command line is different in `master` so please merge master to your branch to be up to date with the master's changes.

Reply:
@commandlinegirl Shall we merge master to nextflow-dev as well?